lib/new_relic/transaction/complete.ex

```diff
@@ -57,6 +58,20 @@ defmodule NewRelic.Transaction.Complete do
     })
   end
 
+  defp derive_queue_duration(%{start_time: start_time, queue_start_us: queue_start_us} = tx)
+       when not is_nil(queue_start_us) do
+    start_time_us = System.convert_time_unit(start_time, :native, :microsecond)
+    queue_duration_us = max(0, start_time_us - queue_start_us)
```

Comment:
This is one protection from skew. This could stay, or the other could stay, or both. I think this one is much more performant since it's not going to clock, so it might be a better candidate to stay.